src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -238,15 +241,53 @@ def validate(s: str) -> str:
             column_value = self.column_joiner.join(sorted(set(column_value)))
             row[column_name] = column_value
 
+    def _get_header(self, object_key):
+        try:
+            header = self.storage_service.head(object_key=object_key)
+        except self.storage_service.client.exceptions.ClientError as e:
+            if int(e.response['Error']['Code']) == 404:
+                return None
+            else:
+                raise e
+        return header
+
+    def _can_use_cached_manifest(self, object_key):
+        """
+        Check if a s3 object was previously created and still exists in the bucket and won't be expiring soon.
+        :param object_key: S3 object key (eg. 'manifests/e0fabf97-7abb-5111-af97-810f1e736c71.tsv'
+        :return: boolean
+        """
+        minimum_remaining_threshold = 60 * 15
+        header = self._get_header(object_key)
+        if header is not None:
+            object_age = datetime.now(timezone.utc) - header['LastModified']
+            seconds_until_expire = config.s3_object_lifetime_seconds - object_age.total_seconds()
+            # if 'x-amz-expiration' in header:
```

Comment:
This is an RFC 2616 HTTP date,  it seems, for which we have `parse_http_date` in `time.py`